model/CSVDialog.cc

```diff
@@ -123,22 +173,35 @@ std::string CSVDialog::loadWebFile(const std::string& url)
   boost::beast::flat_buffer buffer;
 
   // Declare a container to hold the response
-  http::response_parser<http::string_body> res;
-  res.body_limit((std::numeric_limits<std::uint64_t>::max)());
+  http::response<http::dynamic_body> res;
 
   // Receive the HTTP response
   http::read(stream, buffer, res);
-  
-  res.eager(true);  // See https://github.com/boostorg/beast/issues/1352
+
   // Check response status and throw error all values 400 and above. See https://www.boost.org/doc/libs/master/boost/beast/http/status.hpp for status codes
-  if (res.get().result_int() >= 400) throw runtime_error("Invalid HTTP response. Response code: " + std::to_string(res.get().result_int()));
-                                                 
+  if (res.result_int() >= 400) throw runtime_error("Invalid HTTP response. Response code: " + std::to_string(res.result_int()));
+                          
   // Dump the outstream into a temporary file for loading it into Minsky' CSV parser 
   boost::filesystem::path temp = boost::filesystem::unique_path();
   const std::string tempStr    = temp.string();
           
-  std::ofstream outFile(tempStr, std::ofstream::out);  
-  outFile << res.get().body();                                            
+  std::ofstream outFile(tempStr, std::ofstream::binary);  
+  
+  // Handle zipped CSV files. Not working... 
+  if (res[http::field::content_type]=="application/zip") {
+	   
+	  
+	   std::string responseBody = boost::beast::buffers_to_string(res.body().data());
```

Comment:
Check what this does. Usually, strings in C++ are nul-terminated, and this conversion may be corrupting the buffer. If res.body().data() returns a simple const char*, then that is what you should use.

Reply:
res.body().data() returns a boost::beast::basic_multi_buffer which is not convertible to string or char by standard casting. [https://stackoverflow.com/questions/45761865/how-to-convert-boost-beast-multi-buffer-to-string](https://stackoverflow.com/questions/45761865/how-to-convert-boost-beast-multi-buffer-to-string)

Another half-day down the drain. Most of my attempts to add value to this Ravel feature have come to naught. :( 

I've searched high and low and tried several suggestions:

[https://html.developreference.com/article/13439501/Trying+to+understand+the+boost%3A%3Abeast+multibuffer](https://html.developreference.com/article/13439501/Trying+to+understand+the+boost%3A%3Abeast+multibuffer)

and

[https://github.com/boostorg/beast/issues/936](
https://github.com/boostorg/beast/issues/936)

I will have to abandon this endeavour and just implement the fix for ticket 1173....

Reply:
I tried to import the simplest possible zipped csv file at [https://www.stats.govt.nz/large-datasets/csv-files-for-download/](https://www.stats.govt.nz/large-datasets/csv-files-for-download/) :

[https://www.stats.govt.nz/assets/Uploads/Employment-indicators-October-2019-infoshare-tables/monthly-employment-indicators-october-2019.zip](
https://www.stats.govt.nz/assets/Uploads/Employment-indicators-October-2019-infoshare-tables/monthly-employment-indicators-october-2019.zip)

No luck even in this case...